src/Symfony/Component/Security/Http/Authentication/AuthenticatorManagerInterface.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Authentication;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Http\Authenticator\AuthenticatorInterface;
+use Symfony\Component\Security\Http\Firewall\AbstractListener;
+
+/**
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+interface AuthenticatorManagerInterface
+{
+    /**
+     * Convenience method to manually login a user and return a
+     * Response *if any* for success.
+     */
+    public function authenticateUser(UserInterface $user, AuthenticatorInterface $authenticator, Request $request): ?Response;
+
+    /**
+     * Called to see if authentication should be attempted on this request.
+     *
+     * @see AbstractListener::supports()
+     *
+     * @internal
```

Comment:
Not sure it makes sense to have internal methods in an interface: PHP will cry if you forget implementing them, and phpstorm will cry once you implement them :) Right? Should the methods be moved elsewhere, or the flag be removed?

Reply:
The thing is: Nobody should implement this interface (unless they are doing really low level stuff).

A user would typehint this interface in their controller for manual authentication (`authenticateUser()`). I don't want people to use `support()`+`authenticateRequest()` on their own, they should leave that to the `AuthenticatorManagerListener`. However, we would still need to have them in the interface.

Reply:
I've now split the `authenticateUser()` method into another interface, which the user should typehint. That's the safest option. Not happy about the name - `UserAuthenticatorInterface` - but that's something we can always tweak later